modules/islandora_unpaywall/islandora_unpaywall.install

```diff
@@ -0,0 +1,17 @@
+<?php
+/**
+ * @file
+ * Installation hooks for Unpaywall badge.
+ */
+
+/**
+ * Implements hook_uninstall().
+ */
+function islandora_unpaywall_uninstall() {
+  $vars = array(
+    islandora_unpaywall_linktext,
+    islandora_unpaywall_email,
+    islandora_unpaywall_api_endpoint,
+  );
+  array_walk($vars, 'variable_del');
+}
```

Comment:
I saw some discussion of this on the conversation page from @adam-vessey. You could add an `islandora_unpaywall_install()` that checks `if (module_exists("islandora_oadoi")) {` and copies the `islandora_oadoi_linktext` variable to `islandora_unpaywall_linktext`.

Reply:
It seems there was an agreement on doing this on the old module https://github.com/Islandora/islandora_badges/pull/19#issuecomment-390300454 But i agree with Jared that having this in the new one could solve the issue. After a few releases (like islandora 7.x-1.20 😨 ) we could then remove the code

Reply:
Would `islandora_unpaywall_install()` only trigger at the point of first installing the module, or every time it's enabled?

Reply:
First install only, unless they uninstall (not disable) and re-install this module.

Reply:
So I should put `drupal_uninstall_modules(array('islandora_oadoi'));` in `islandora_unpaywall.install`, after `islandora_unpaywall_install()` copies the variable?

Reply:
You can, just be sure to output information to the screen that you are doing that. People can be confused when one module uninstalls another one 😄

Reply:
Message added with latest commit.